File: backend/api/itinerary.py
"""
行程API
"""
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
from typing import Optional, Dict, Any, List

router = APIRouter()


class ItineraryRequest(BaseModel):
    """行程生成请求"""
    session_id: str
    requirements: Dict[str, Any]


class ItineraryResponse(BaseModel):
    """行程响应"""
    success: bool
    data: Optional[Dict[str, Any]] = None
    error: Optional[str] = None


@router.post("/itinerary/generate")
async def generate_itinerary(request: ItineraryRequest) -> ItineraryResponse:
    """
    生成行程
    """
    try:
        # TODO: 调用行程生成器
        return ItineraryResponse(
            success=True,
            data={
                "itinerary_id": "example_id",
                "markdown": "# 示例行程",
                "daily_plans": [],
                "total_budget": 0
            }
        )
    except Exception as e:
        return ItineraryResponse(
            success=False,
            error=str(e)
        )


@router.get("/itinerary/{itinerary_id}")
async def get_itinerary(itinerary_id: str) -> ItineraryResponse:
    """
    获取行程详情
    """
    try:
        # TODO: 从数据库查询
        return ItineraryResponse(
            success=True,
            data={}
        )
    except Exception as e:
        return ItineraryResponse(
            success=False,
            error=str(e)
        )


@router.put("/itinerary/{itinerary_id}")
async def update_itinerary(
    itinerary_id: str,
    updates: Dict[str, Any]
) -> ItineraryResponse:
    """
    修改行程
    """
    try:
        # TODO: 更新行程
        return ItineraryResponse(
            success=True,
            data={"message": "行程已更新"}
        )
    except Exception as e:
        return ItineraryResponse(
            success=False,
            error=str(e)
        )

File: frontend/src/api/itinerary.js
import request from './index'

/**
 * 生成行程
 */
export function generateItinerary(data) {
  return request({
    url: '/itinerary/generate',
    method: 'post',
    data
  })
}

/**
 * 获取行程详情
 */
export function getItinerary(itineraryId) {
  return request({
    url: `/itinerary/${itineraryId}`,
    method: 'get'
  })
}

/**
 * 更新行程
 */
export function updateItinerary(itineraryId, data) {
  return request({
    url: `/itinerary/${itineraryId}`,
    method: 'put',
    data
  })
}

